### ISSUE TITLE
Creating Private Repo with Twitter Data to Use With Codespaces

### ISSUE THREAD

**Issue by sking115422** [quinngroup/Twitter-Embedding-Analysis#37]
We would like to start using codespaces for testing. However, we have realized that we need to have the twitter data on the repo to run the code. I understand there may be some privacy issues with this. Can we create a private repo and clone our current repo into that one for testing with code spaces to protect the integrity of the data?

There may also be a better alternative I am not thinking of. Please let us know how you would like us to handle this.

**Comment by magsol** [comment]
From what I've heard from others (working on the @quinngroup/ciliaweb-platform project), there are issues associated with private repos + Codespaces. I have not actually tried this, but: my inclination would be to stick with an open repo (for numerous unrelated reasons).

To that end, if you could create a series of datasets (a new ticket for this might be good) that **are all under 80MB or so**, those could definitely be added to the repo.

Point being, we don't want the repo to be too heavy on initial clone, nor do we want to break general versioning convention by storing large static files in the repo. However, a solid argument could be made for e.g. small, medium, and "large" example / processed datasets (e.g., 25MB, 50MB, and 75MB).

Key point is "processed": another issue to create would be how to process the data so we can still work meaningfully with it, while also not running afoul of Twitter TOS. I could imagine this might be as simple as performing one-way hashes (e.g. sha256) on usernames; that way they still map the same hash to the same username, but it's otherwise anonymized. We could look at other methods too.

tl;dr Yes but keep the repo open, meaning we need some additional tickets.

**Comment by sking115422** [comment]
Gotcha. I will definitely start looking in to that. I will try to have some smaller processed datasets ready in the next couple weeks, and encrypt all of the sensitive info so that we can start putting Codespaces to use. 

Thanks for the insight on this.
